frontend/src/components/common/SideBar.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { FC, ReactNode } from "react";
+import { useNavigate } from "react-router-dom";
+import {
+  Tabs,
+  TabList,
+  Tab,
+  Box,
+  Avatar,
+  Text,
+  Flex,
+  Heading,
+  useBreakpointValue,
+} from "@chakra-ui/react";
+
+const SideBar: FC<{ children: ReactNode }> = () => {
+  const navigate = useNavigate();
+
+  const handleTabClick = (path: string) => {
+    navigate(path);
+  };
+
+  const pages = [
+    { label: "Home", route: "/home" },
+    { label: "Tasks", route: "/tasks" },
+    { label: "Approvals", route: "/approvals" },
+    { label: "Schedule", route: "/schedule" },
+    { label: "Participants", route: "/participants" },
+    { label: "Insights", route: "/insights" },
+  ];
+
+  const sidebarWidth = useBreakpointValue({
+    base: "100%",
+    md: "50%",
+    lg: "30%",
+    xl: "20%",
+  });
+
+  return (
+    <Flex flexDir="column" w={sidebarWidth} height="calc">
+      <Box
+        h="calc(100vh)"
+        borderRight="solid"
+        borderRightColor="grey"
```

Comment:
For this colour, I would prefer if you used the grey that you defined in the Chakra theme (as long as it still looks good).